src/components/Toast/ToastContext.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { createContext, PropsWithChildren, useContext, useMemo, useState } from "react";
+import { Toast } from "./Toast";
+import { ToastNoticeProps } from "./ToastNotice";
+
+export type ToastContextProps = {
+  setNotice: React.Dispatch<React.SetStateAction<ToastNoticeProps>>;
+};
+
+export const ToastContext = createContext<ToastContextProps>({
+  setNotice: () => {},
```

Comment:
Suggestion: Could make this:

```
setNotice: () => throw new Error("Missing ToastProvider")
```

Just to more explicitly "fail fast" if someone tries to `showNotice` while missing the provider.